File: arrays/check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.ts
function check(nums: number[]): boolean {
  let count: number = 0;
  let n: number = nums.length;

  for (let i = 0; i < n; i++) {
    let next = nums[(i + 1) % n];

    if (nums[i] > next) {
      count++;
    }

    if (count > 1) {
      return false;
    }
  }
  return true;
}

File: arrays/check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.go
func check(nums []int) bool {
	count := 0
	n := len(nums)

	for i := 0; i < n; i++ {
		next := nums[(i+1)%n]
		if nums[i] > next {
			count++
		}

		if count > 1 {
			return false
		}
	}
	return true
}
